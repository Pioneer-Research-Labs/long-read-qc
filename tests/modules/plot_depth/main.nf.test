nextflow_process {

    name "Test Process plot_depth"
    script "modules/plot_depth/main.nf"
    process "plot_depth"

    test("Should plot sequencing depth"){

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam.bai'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts_stats.tsv')
                )])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            ("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/coverage_plot.png")
            ).match()
        }

    }

}
