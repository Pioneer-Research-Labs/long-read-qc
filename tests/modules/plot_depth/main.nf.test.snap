{
    "Should plot sequencing depth": {
        "content": [
            "coverage_plot.png:md5,ffa08f1428caf1d68ecaeacb04203e67"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T13:27:28.968495"
    }
}